ESP_ATMod/ESP_ATMod.ino

```diff
@@ -632,10 +752,46 @@ int SendData(int clientIndex, int maxSize)
  * Returns the internal char* of the input string
  * In case of empty string returns nullptr
  */
-const char* nullIfEmpty(String& s)
+const char *nullIfEmpty(String &s)
 {
 	if (s.isEmpty())
 		return nullptr;
 
 	return s.c_str();
 }
+
+/*
+ * Checks if the newly added certificate is a duplicate
+ */
+bool checkCertificateDuplicates(size_t indexNewAdded)
+{
+	const br_x509_certificate *certToCheck = &(CAcert.getX509Certs()[indexNewAdded]);
+
+	for (size_t i = 0; i < CAcert.getCount(); i++)
+	{
+		const br_x509_certificate *cert2 = &(CAcert.getX509Certs()[i]);
+		if (!memcmp(certToCheck->data, cert2->data, sizeof(certToCheck->data)) && i != indexNewAdded)

```

Comment:
Is `sizeof(certToCheck->data)` correct? 
Shouldn't we test 
`certToCheck->data_len == cert2->data_len` 
and then  
`memcmp(certToCheck->data, cert2->data, certToCheck->data_len)` ?

Reply:
Resolved.